### PR TITLE
Add SEGGER Systemview support

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
### Solution
Add support for SEGGER Systemview
https://www.segger.com/products/development-tools/systemview/

Sample config for MR-CANHUBK3 
- nxp_mr-canhubk3_sysview

### Alternatives
Right I'm patching sched_sysview.c using cmake to make it co-exist nicely with cpuload.cpp not sure what would be a better solution.

Edit: I've changed it now to use objcopy to changes the symbol names of sched_sysview which is a bit better.

### Test coverage
Tested on nxp_mr-canhubk3_sysview
Autodetect of SEGGER RTT Control Block Address doesn't work have to manually specify based on boot print

### Context
image

### CODE REVIEW

File: platforms/nuttx/NuttX/CMakeLists.txt [hunk 1]
```diff
@@ -180,10 +179,48 @@ function(add_nuttx_dir nuttx_lib nuttx_lib_dir kernel extra target)
 	add_dependencies(nuttx_${nuttx_lib} nuttx_${nuttx_lib}_build)
 endfunction()
 
+# helper for drivers with segger_sysview patches
+function(add_nuttx_dir_patched nuttx_lib nuttx_lib_dir kernel extra target)
```
Comment (dagar) [inline platforms/nuttx/NuttX/CMakeLists.txt:183]: Maybe a bit more specific naming? I had to dig in to spot the segger differences.

```suggestion
function(add_nuttx_dir_segger_patched nuttx_lib nuttx_lib_dir kernel extra target)
```

File: platforms/nuttx/NuttX/CMakeLists.txt [hunk 2]
```diff
@@ -180,10 +179,48 @@ function(add_nuttx_dir nuttx_lib nuttx_lib_dir kernel extra target)
...
+	file(GLOB_RECURSE nuttx_lib_files LIST_DIRECTORIES false
+		${CMAKE_CURRENT_SOURCE_DIR}/nuttx/${nuttx_lib_dir}/*.c
+		${CMAKE_CURRENT_SOURCE_DIR}/nuttx/${nuttx_lib_dir}/*.h
+	)
+
+	add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/nuttx/${nuttx_lib_dir}/lib${nuttx_lib}.a
+		COMMAND ${CMAKE_COMMAND} -E remove -f ${NUTTX_DIR}/${nuttx_lib_dir}/lib${nuttx_lib}.a
+		COMMAND find ${nuttx_lib_dir} -type f -name \*.o -delete
+		COMMAND make -C ${nuttx_lib_dir} --no-print-directory --silent ${nuttx_lib_target} TOPDIR="${NUTTX_DIR}" KERNEL=${kernel} EXTRAFLAGS=${extra}
+		COMMAND ${CMAKE_OBJCOPY} ${NUTTX_DIR}/${nuttx_lib_dir}/lib${nuttx_lib}.a --redefine-sym sched_note_start=segger_note_start --redefine-sym sched_note_stop=segger_note_stop --redefine-sym sched_note_suspend=segger_note_suspend --redefine-sym sched_note_resume=segger_note_resume ${NUTTX_DIR}/${nuttx_lib_dir}/lib${nuttx_lib}.a
```
Comment (dagar) [inline platforms/nuttx/NuttX/CMakeLists.txt:199]: Can we simply have different scheduling notes built based on CONFIG_SEGGER_SYSVIEW or is there a reason it needs to be patched like this?

Comment (PetervdPerk-NXP) [reply platforms/nuttx/NuttX/CMakeLists.txt:199]: So both cpuload and sysview use the NuttX note hooks, therefore functions like `sched_note_start` are duplicate. The trick what I do here is too rename the `sched_note_start` from sysview into `segger_note_start` and call function from the cpuload `sched_note_start` to make sure that both PX4 cpuload works and sysview.

Comment (dagar) [reply platforms/nuttx/NuttX/CMakeLists.txt]: Makes sense, I was just wondering if we could do it without duplicating this big ugly wad of cmake.

Comment (PetervdPerk-NXP) [reply platforms/nuttx/NuttX/CMakeLists.txt]: Well @davids5 suggested another solution to use the linker wrap functionality to wrap the `sched_note_xxx` instances during linking process and then we just call the wrapped function from cpuload.cpp. I've tried this but linking in the PX4 CMake is rather complex.

**Comment by davids5** [comment]
@PetervdPerk-NXP Why don't we upstream a Kconfig Prefix that will be added to the the system view code's function name to removed the conflict. The default can be "".

**Comment by PetervdPerk-NXP** [comment]
B

> @PetervdPerk-NXP Why don't we upstream a Kconfig Prefix that will be added to the the system view code's function name to removed the conflict. The default can be "".

Could be a solution although I doubt they would accept it since it would serve no purpose in NuttX.

@dagar I've cleaned up the cmake a bit now it's a bit cleaner by using `COMMAND_EXPAND_LISTS` instead of a seperate function.

**Comment by davids5** [comment]
> Could be a solution although I doubt they would accept it since it would serve no purpose in NuttX.

I think it It is a fair ask for the same reason we have, and that sort of thing has been accepted in the past.

**Review by davids5 — DISMISSED** [review]
LGTM

**Comment by davids5** [comment]
@PetervdPerk-NXP please  ensure apps at a489381b49835ecba6f3b873b5071d882a18152f

**Comment by davids5** [comment]
> @PetervdPerk-NXP please ensure apps at a489381b49835ecba6f3b873b5071d882a18152f (current px4_firmware_nuttx-10.3.0+)

Also Please rebase on 1ee8fb827c570f302a43ca9870c58d1ac73997d3 (current nuttx px4_firmware_nuttx-10.3.0+)

**Review by davids5 — APPROVED** [review]
LGTM

**Pull request merged**